astropy/units/quantity.py

```diff
@@ -666,8 +679,20 @@ def _to_value(self, unit, equivalencies=[]):
         """Helper method for to and to_value."""
         if equivalencies == []:
             equivalencies = self._equivalencies
-        return self.unit.to(unit, self.view(np.ndarray),
-                            equivalencies=equivalencies)
+        if not self.dtype.names or isinstance(self.unit, StructuredUnit):
+            # Standard path, let unit to do work.
+            return self.unit.to(unit, self.view(np.ndarray),
+                                equivalencies=equivalencies)
+
+        else:
+            # The .to() method of a simple unit cannot convert a structured
+            # dtype, so we work around it, by recursing.
+            # TODO: deprecate this?
+            # Convert simple to Structured on initialization?
```

Comment:
``# Convert simple to Structured on initialization?``
as in always work with structured units?

Reply:
Yes, the suggestion is that for structured arrays, the unit should always be structured too. As is, it never worked very well. But `EarthLocation` uses it, so it would be nice not to mess things up too much. My sense is that it is best for follow-up (but ideally before 5.0 is out...)